integrations/dns_data.go

```diff
@@ -0,0 +1,162 @@
+package integrations
+
+import (
+	"errors"
+	cf "github.com/cloudflare/cloudflare-go"
+	"time"
+)
+
+type DNSResource struct{
+	Zone string
+	Name string
+	cfRecordId string
+	cfZoneId string
+	cfProxied bool
+	cfModifiedOn time.Time
+}
+
+func (f DNSResource)FQDN()string{
+	return f.Name + "." + f.Zone
+}
+
+func (f *DNSResource)translateFromCloudflare(record cf.DNSRecord){
+	if f==nil{
+		*f = DNSResource{}
+	}
+	f.Zone=record.ZoneName
+	f.Name=record.Name
+	f.cfRecordId=record.ID
+	f.cfZoneId=record.ZoneID
+	f.cfProxied=record.Proxied
+	f.cfModifiedOn=record.ModifiedOn
+}
+
+
+type Record struct {
+	Resource  DNSResource
+	Ttl       uint
+	DnsType   string
+	DnsData   string
```

Comment:
Codacy found an issue: [struct field DnsData should be DNSData](https://app.codacy.com/gh/go-go-dojo/friendly-potato/pullRequest?prid=6165294)